File: src/domain/models/stream_test.go
package models

import (
	"testing"

	"Theatrum/constants"
)

func TestStream_GetMasterPlaylistTemplatePath(t *testing.T) {
	tests := []struct {
		name   string
		path   string
		expect string
	}{
		{
			"simple path",
			"videos/{username}",
			"videos/{username}/" + constants.MasterPlaylist,
		},
		{
			"nested path",
			"live/{room_id}/{user}",
			"live/{room_id}/{user}/" + constants.MasterPlaylist,
		},
		{
			"path with builtin function",
			"recordings/{%STARTING_DATE%}",
			"recordings/{%STARTING_DATE%}/" + constants.MasterPlaylist,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			s := &Stream{Path: tt.path}
			got := s.GetMasterPlaylistTemplatePath()
			if got != tt.expect {
				t.Errorf("GetMasterPlaylistTemplatePath() = %q, want %q", got, tt.expect)
			}
		})
	}
}

func TestStream_GetDashManifestTemplatePath(t *testing.T) {
	tests := []struct {
		name   string
		path   string
		expect string
	}{
		{
			"simple path",
			"videos/{username}",
			"videos/{username}/" + constants.DashManifest,
		},
		{
			"nested path",
			"live/{room_id}/{user}",
			"live/{room_id}/{user}/" + constants.DashManifest,
		},
		{
			"path with builtin function",
			"recordings/{%STARTING_DATE%}",
			"recordings/{%STARTING_DATE%}/" + constants.DashManifest,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			s := &Stream{Path: tt.path}
			got := s.GetDashManifestTemplatePath()
			if got != tt.expect {
				t.Errorf("GetDashManifestTemplatePath() = %q, want %q", got, tt.expect)
			}
		})
	}
}

File: src/test/java/com/fjourdren/theatrum/domain/model/StreamTest.java
package com.fjourdren.theatrum.domain.model;

import com.fjourdren.theatrum.domain.constant.VideoConstants;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.Assertions.assertThat;

class StreamTest {

    @Test
    void restreamTypeValue() {
        assertThat(StreamType.RESTREAM.value()).isEqualTo("restream");
    }

    @ParameterizedTest(name = "{0}")
    @CsvSource({
            "simple path,               videos/{username}",
            "nested path,               live/{room_id}/{user}",
            "path with builtin function,recordings/{%STARTING_DATE%}"
    })
    void getMasterPlaylistTemplatePath(String name, String path) {
        Stream stream = Stream.builder().path(path).build();
        assertThat(stream.getMasterPlaylistTemplatePath()).isEqualTo(path + "/" + VideoConstants.MASTER_PLAYLIST);
    }

    @ParameterizedTest(name = "{0}")
    @CsvSource({
            "simple path,               videos/{username}",
            "nested path,               live/{room_id}/{user}",
            "path with builtin function,recordings/{%STARTING_DATE%}"
    })
    void getDashManifestTemplatePath(String name, String path) {
        Stream stream = Stream.builder().path(path).build();
        assertThat(stream.getDashManifestTemplatePath()).isEqualTo(path + "/" + VideoConstants.DASH_MANIFEST);
    }
}
